File: Client/Tuckshop.WebApp/src/Identity/Models/UserManagement/UserManagementAction.ts
import { EnumHelper } from '@singularsystems/neo-core';

export enum UserManagementAction {
    ResendEmailVerificationLink,
    ResetMFA,
    ClearLockout,
    Activate,
    Deactivate,
    EnableMFA,
    DisableMFA,
}

EnumHelper.decorateEnum(UserManagementAction, decorator => {
    decorator.describe(UserManagementAction.ResendEmailVerificationLink, "ResendEmailVerificationLink", "Resend the email verification link for {User}");
    decorator.describe(UserManagementAction.ResetMFA, "ResetMFA", "Reset MFA for {User}");
    decorator.describe(UserManagementAction.ClearLockout, "ClearLockout", "Clear lockout for {User}");
    decorator.describe(UserManagementAction.Activate, "Activate", "Activate {User}");
    decorator.describe(UserManagementAction.Deactivate, "Deactivate", "Deactivate {User}");
    decorator.describe(UserManagementAction.EnableMFA, "EnableMFA", "Enable MFA for {User}");
    decorator.describe(UserManagementAction.DisableMFA, "DisableMFA", "Disable MFA for {User}");
});

File: Server/Tuckshop.IdentityServer/Tuckshop.IdentityServer.Contracts/Models/UserManagement/UserManagementAction.cs
#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
namespace Tuckshop.IdentityServer.Contracts.UserManagement
{
  using System.ComponentModel.DataAnnotations;
  using Neo.Model.DataAnnotations;

  /// <summary>
  /// Enum representing the types of user management actions
  /// </summary>
  [DbEnum]
  public enum UserManagementAction
  {
    [Display(Description = "Resend the email verification link for {User}")]
    ResendEmailVerificationLink,

    [Display(Description = "Reset MFA for {User}")]
    ResetMFA,

    [Display(Description = "Clear lockout for {User}")]
    ClearLockout,

    [Display(Description = "Activate {User}")]
    Activate,

    [Display(Description = "Deactivate {User}")]
    Deactivate,

    [Display(Description = "Enable MFA for {User}")]
    EnableMFA,

    [Display(Description = "Disable MFA for {User}")]
    DisableMFA,
  }
}
